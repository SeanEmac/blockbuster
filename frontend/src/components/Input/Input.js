import React from "react";
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';

const Input = (props)  => {
  return (
    <React.Fragment>
      <Typography component="h2" variant="h6" color="primary" gutterBottom>
        Search for a transaction:
      </Typography>
      <form onSubmit={e => props.getTransactionID(e, e.target.transactionID.value)}>
        <TextField
          id = "filled-full-width"
          label = "Transaction ID"
          name = "transactionID"
          style = {{ margin: 8 }}
          defaultValue = "b6f6991d03df0e2e04dafffcd6bc418aac66049e2cd74b80f14ac86db1e3f0da"
          helperText = "Enter a Bitcoin transaction ID"
          fullWidth
          margin = "normal"
          InputLabelProps = {{
            hrink: "true",
          }}
        />
      </form>
    </React.Fragment>
  )
}

export default Input;