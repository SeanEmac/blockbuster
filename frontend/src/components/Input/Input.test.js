import React from 'react';
import Input from './Input';
import { shallow } from 'enzyme';
import TextField from '@material-ui/core/TextField';

describe('Transaction input component', () => {
  const getTransactionID = jest.fn()
  const component = shallow(<Input getTransactionID={getTransactionID}/>);

  it('should match the snapshot', () => {
    expect(component.html()).toMatchSnapshot();
  });

  it('propegates transaction ID on submition', () => {
    expect(component.find('form').length).toBe(1);
    expect(component.find(TextField).length).toBe(1);

    component.find('form').simulate('submit', {
      target: {transactionID: {value: 'My new value'}}
    });

    expect(getTransactionID).toHaveBeenCalledWith(
      {"target": { "transactionID": {"value": "My new value"}}},
      "My new value"
    );
  })
});