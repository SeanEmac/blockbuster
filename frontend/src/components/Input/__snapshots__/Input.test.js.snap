// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction input component should match the snapshot 1`] = `"<h2 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom\\">Search for a transaction:</h2><form><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\" style=\\"margin:8px\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\" for=\\"filled-full-width\\" id=\\"filled-full-width-label\\" hrink=\\"true\\">Transaction ID</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" aria-describedby=\\"filled-full-width-helper-text\\" id=\\"filled-full-width\\" name=\\"transactionID\\" value=\\"b6f6991d03df0e2e04dafffcd6bc418aac66049e2cd74b80f14ac86db1e3f0da\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div><p class=\\"MuiFormHelperText-root MuiFormHelperText-filled\\" id=\\"filled-full-width-helper-text\\">Enter a Bitcoin transaction ID</p></div></form>"`;
